// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option objc_class_prefix = "AU";

package user;

service User {
  rpc GetUser (GetUserRequest) returns (GetUserReply) {}
  rpc SetUser (SetUserRequest) returns (SetUserReply) {}
}

message GetUserRequest {
  int32 id = 1;
}

message SetUserRequest {
  string first_name = 1;
  string last_name  = 2;
  string email      = 3;
  string title      = 4;
  string login      = 5;
  string password   = 6;
}

message GetUserReply {
  int32 id = 1;
  string first_name = 2;
  string last_name  = 3;
  string email      = 4;
  string title      = 5;
  string login      = 6;
  string password   = 7;
}

message SetUserReply {
  int32 id = 1;
}